site/docs/components/drawer.mdx

```diff
@@ -0,0 +1,128 @@
+*   When you expect this functionality to be required on mobile. Because a small drawer does not need to rearrange or re-flow like a page or modal does, it is often a friendlier experience on mobile. 
+*   When you can structure content using just headings or [collapsible rows](/components/collapsible/) (i.e. you do not require additional navigation like tabs). 
+
+### Overlay vs Parallel
+
+Drawers can have overlays or can appear in parallel with page content. 
+
+**Overlay drawers:**
+*   Are useful when you want to focus the user on a specific task. 
+*   Sit over the top of page content, like a [modal](/components/modal), meaning a user is unable to interact with the main page while the drawer is open. Unlike a modal, clicking outside the drawer should not close the drawer. This is to prevent the user losing unsaved work. 
+*   Are full viewport height (covering the main nav). 
+*   Are vertically scrollable separate to the main page content. When the overlay is open, you can only scroll inside the overlay (i.e. when you scroll to the bottom of the drawer,the rest of the page should not start scrolling).
+
+**Parallel drawers:**
+*   Push page content to the left and display in parallel, with no overlay. This is useful for providing additional context or information during a task. 
+*   Allow users to continue interacting with the main page content, including clicking links. Navigating away from the page will close the drawer. For this reason, parallel drawers should be avoided when asking the user to input information or make progress that will be lost by closing the page. 
+*   Are vertically scrollable separate to the main page content. 
+*   Start below the main header navigation so navigation elements are still available. 
+
+### Headers and footers
+
+A drawer always has a header that consists of:
+*   A left aligned, short H2 title (<35 characters for a medium drawer, <15 characters for a small drawer)
+*   A Close X button, positioned on the left hand side
+*   A full-width horizontal [divider](/components/divider/) to separate the header from body content
+
+If the Drawer involves a [primary action](/components/button/) like a Save function, it should be housed within a footer. 
+
+If the content within the drawer exceeds the screen height, the header (and footer, if applicable) should be fixed and always visible. 
+
+### Dismissing/closing a Drawer
+
+*   Overlay drawers are not dismissed by clicking outside the window (unlike modals). This is to prevent users accidentally losing their work. 
+
+Users can dismiss a Drawer by:
+*   Clicking an X close button
+*   Pressing the Esc key
+*   Clicking the primary action (e.g. Save or Submit) 
+*   Upon dismissal, the drawer focus should be restored to the triggering element
+
+### Mobile/Responsiveness
+
+*   A Small Drawer maintains the same behavior all the way down to viewports of 300px, at which point it begins taking up 100% of the viewport width.
```

Comment:
I'm confused here. Wouldn't it start taking up 100% of the viewport width at 375px if the drawer is 375px wide? I'd suggest something more like "all the way down to viewports of 405px" (375px + 20px). Because if you have a screen that's 380px and the drawer is 375px wide, you'll get a very narrow (and ugly?) 5px gap to the left, so it might be better to fill the whole screen at screen widths less than about ~400px.